### PR TITLE
Cleanup segment upload logic and allow validation on real-time table

### PR DESCRIPTION
**Pull request by Jackie-Jiang**
Subtask of #8283 

- Cleanup the segment upload logic and add more checks on the parameters
- Refactor the segment validator to a util class to make it work on both metadata push and real-time table
- Refine the return code documentation of the segment upload API

### CODE REVIEW

**Review by sajjad-moradi — APPROVED** [review]
LGTM

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentUploadDownloadRestletResource.java [hunk 1]
```diff
@@ -253,67 +284,70 @@ private SuccessResponse uploadSegment(@Nullable String tableName, TableType tabl
...
+      if (StringUtils.isNotEmpty(tableName)) {
         rawTableName = TableNameBuilder.extractRawTableName(tableName);
-        LOGGER.info("Uploading a segment {} to table: {}, push type {}, (Derived from API parameter)", segmentName,
-            tableName, uploadType);
       } else {
         // TODO: remove this when we completely deprecate the table name from segment metadata
         rawTableName = segmentMetadata.getTableName();
-        LOGGER.info("Uploading a segment {} to table: {}, push type {}, (Derived from segment metadata)", segmentName,
-            tableName, uploadType);
+        LOGGER.warn("Table name is not provided when uploading segment: {} for table: {}", segmentName, rawTableName);
```
Comment (jackjlli) [inline pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentUploadDownloadRestletResource.java:292]: Make it more explicit like: 
`Table name is not provided when uploading segment: {} for table: {} from segment metadata"`?

**Review by jackjlli — APPROVED** [review]
Overall LGTM.

File: pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentUploadDownloadRestletResource.java (hunk 1 above)
Comment (Jackie-Jiang) [reply pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentUploadDownloadRestletResource.java:292]: We log this warning when it is not provided through the query param of the rest request. Filed a follow up PR: #8715

**Pull request merged**